### PR TITLE
Kfold

### PR DESCRIPTION
**Pull request by VolodymyrOrlov**
This PR adds new interface, `Predictor`, that defines the only method `predict`.
I then refactor all of the supervised learning algorithms to unify layout of method `fit`. 

Finally, I add new function, `cross_validate`, that performs k-fold cross validation for a given estimator

### CODE REVIEW

File: src/naive_bayes/gaussian.rs [hunk 1]
```diff
@@ -85,6 +86,11 @@ impl<T: RealNumber> GaussianNBParameters<T> {
     pub fn new(priors: Option<Vec<T>>) -> Self {
         Self { priors }
     }
+    /// Prior probabilities of the classes. If specified the priors are not adjusted according to the data
+    pub fn with_priors(mut self, priors: Vec<T>) -> Self {
```
Comment (morenol) [inline src/naive_bayes/gaussian.rs:90]: maybe we can remove `pub fn new`

File: src/naive_bayes/bernoulli.rs [hunk 2]
```diff
@@ -95,6 +96,21 @@ impl<T: RealNumber> BernoulliNBParameters<T> {
             binarize,
         }
     }
+    /// Additive (Laplace/Lidstone) smoothing parameter (0 for no smoothing).
+    pub fn with_alpha(mut self, alpha: T) -> Self {
```
Comment (morenol) [inline src/naive_bayes/bernoulli.rs:100]: Maybe we can remove the `pub fn new`)

File: src/naive_bayes/multinomial.rs [hunk 3]
```diff
@@ -85,6 +86,16 @@ impl<T: RealNumber> MultinomialNBParameters<T> {
     pub fn new(alpha: T, priors: Option<Vec<T>>) -> Self {
         Self { alpha, priors }
     }
+    /// Additive (Laplace/Lidstone) smoothing parameter (0 for no smoothing).
+    pub fn with_alpha(mut self, alpha: T) -> Self {
```
Comment (morenol) [inline src/naive_bayes/multinomial.rs:90]: maybe we can remove the `pub fn new`

**Review by morenol — APPROVED** [review]
lgtm, only a few suggestions

**Review by rabbitrabid — APPROVED** [review]

**Review by morenol — APPROVED** [review]

**Pull request merged**